src/containerapp/azext_containerapp/_up_utils.py

```diff
@@ -138,6 +149,145 @@ def check_exists(self):
         return self.exists
 
 
+class Extension(Resource):
+    def __init__(
+            self,
+            cmd,
+            name=None,
+            exists: bool = None,
+            namespace=None,
+            logs_location=None,
+            logs_rg=None,
+            logs_customer_id=None,
+            logs_share_key=None,
+            connected_cluster_id=None,
+            connected_environment_name=None,
+    ):
+        resource_group = ResourceGroup(cmd, logs_rg, logs_location)
+        super().__init__(cmd, name, resource_group, exists)
+        self.exists = exists
+        self.namespace = namespace
+        self.logs_location = logs_location
+        self.logs_rg = logs_rg
+        self.logs_customer_id = logs_customer_id
+        self.logs_share_key = logs_share_key
+        self.connected_cluster_id = connected_cluster_id
+        self.connected_environment_name = connected_environment_name
+
+    def create(self):
+        extension = create_extension(cmd=self.cmd,
```

Comment:
why not pass the name of extension?
And should you also check if self.name is None here which is same as create_if_needed()

Reply:
Yes. I will fix that.